src/Network/Minio/Data.hs

```diff
@@ -95,8 +101,106 @@ data ConnectInfo = ConnectInfo {
 -- | Connects to a Minio server located at @localhost:9000@ with access
 -- key /minio/ and secret key /minio123/. It is over __HTTP__ by
 -- default.
-instance Default ConnectInfo where
-  def = ConnectInfo "localhost" 9000 "minio" "minio123" False "us-east-1" True
+-- instance Default ConnectInfo where
+--   def = ConnectInfo "localhost" 9000 "minio" "minio123" False "us-east-1" True
+
+
+defaultConnectInfo :: ConnectInfo
+defaultConnectInfo = ConnectInfo "localhost" 9000 "minio" "minio123" False "us-east-1" True
+
+instance IsString ConnectInfo where
+    fromString str = let req = NC.parseRequest_ str
+                     in defaultConnectInfo { connectHost = TE.decodeUtf8 $ NC.host req
+                                           , connectPort = NC.port req
+                                           , connectIsSecure = NC.secure req
+                                           }
+
+getAwsCreds :: IO (Maybe (Text, Text))
```

Comment:
if there is an attempt to do this, I would prefer if you can follow the https://github.com/minio/minio-go/tree/master/pkg/credentials style where we will simply implement many variations of the same credentials.Value structure. 

This would allow using minio-hs in a wide range of situations. Maybe not everything but starting with aws creds, iam and environment variables

Reply:
Link for the module docs: https://godoc.org/github.com/minio/minio-go/pkg/credentials